### PR TITLE
Python scripts

### PR DESCRIPTION
**Pull request by steven-murray**
This is Boom's PR really, I'm just creating it to make it easier to review. The idea is to add better scripts for running and managing the simulations.

### CODE REVIEW

File: config_files/tele_config_hera_dipole_cubic.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+beam_paths:
+  0: '/ilifu/astro/projects/hera/Validation/h4c_sim/config_files/NF_HERA_Dipole_efield_beam.fits'
```
Comment (steven-murray) [inline config_files/tele_config_hera_dipole_cubic.yaml:2]: this is not computer-agnostic. I recommend having a little script that 'makes' the tele config, and knows the path to the beam file for that computer. Makes it easier to set the frequency/spatial interpolation as well.

File: hpc_config.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  nodes: 1  # Number of nodes
+  mem: "8GB"  # Memory with unit
+  ntasks: 1  # Number of tasks
+  cpus-per-task: 2  # Number of CPUs per task
+  time: "0-00:30:00"  # slurm-aware time string format
+  gpus: 1  # Number of GPU
+  gres: "gpu:1"  # Some cluster do not support --gpus flag
+  # Job name, will be modified to <sky_model>-fch<freq_chan>-chunk<chunk> 
+  # when running H4C sim via run_sim.py
+  job-name: "hera-sim"
```
Comment (steven-murray) [inline hpc_config.yaml:24]: why have the job name it's just overwritten?

File: make_obsparams.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    obsparams_dir = CFGDIR / f"obsparams/{sky_model}/nt17280_spl{chunks}"
+    obsparams_dir.mkdir(parents=True, exist_ok=True)
+
+    for fch, fv in zip(freq_chans, freq_vals):
+        for ch in range(chunks):
+            # Note that global paths from utils are Path object. f-string formatting
+            # automatically converty them to string for yaml to write out.
+            obsparams = {
+                "filing": {
+                    "outdir": f"{OUTDIR}/{sky_model}/nt17280",
```
Comment (steven-murray) [inline make_obsparams.py:94]: I think we should additionally ensure that the file paths are absolute, so `OUTDIR.absolute()`

File: run_sim.py [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+                        )
+                        # Write job script and submit
+                        # Maybe we should save each job script individually
+                        with open("_sim.sbatch", "w") as fl:
+                            fl.write(job_script)
+
+                        subprocess.call(
+                            "sbatch _sim.sbatch".split()
+                        ) if not dry_run else None
+                        logger.info(f"\n===Job Script===\n{job_script}\n===END===\n")
```
Comment (steven-murray) [inline run_sim.py:215]: Yes, I would create a new directory, something like `batch_scripts/` and write each sbatch file into it, with the same name as the job name. That enables checking later.

**Review by steven-murray — COMMENTED** [review]
Thanks @piyanatk, this will make things a *lot* nicer in the future. My only other request other than the few small things I commented on, is to add a README which points us to the overview of how to use the scripts, which ones to use, etc. It'd also be fantastic if the sky models could be created on-the-fly like the obsparams.

File: config_files/tele_config_hera_dipole_cubic.yaml (hunk 1 above)
Comment (piyanatk) [reply config_files/tele_config_hera_dipole_cubic.yaml:2]: sounds good. maybe another  file call h4c_sim_info.yaml or something like that

File: hpc_config.yaml (hunk 2 above)
Comment (piyanatk) [reply hpc_config.yaml:24]: for custom sims. there is a cli function in the main script that has not been written. the idea there is to allow custom simulation where the user must specify everything. the job-name should be customizable for that

File: make_obsparams.py (hunk 3 above)
Comment (piyanatk) [reply make_obsparams.py:49]: They are absolute paths. I declare variables for the simulation paths in `utils.py` and they are resolved there.

File: config_files/tele_config_hera_dipole_cubic.yaml (hunk 1 above)
Comment (steven-murray) [reply config_files/tele_config_hera_dipole_cubic.yaml:2]: Yeah, something like that.

**Review by steven-murray — COMMENTED** [review]
I approve (but I can't formally approve...)

**Pull request merged**